content/references/markdown/extensions.md

```diff
@@ -0,0 +1,130 @@
+---
+kind: reference
+topic: markdown
+prev: /references/markdown/
+---
+
+# Codewars' Markdown Extensions
+
+Codewars adds few Markdown extensions for writing kata descriptions.
+
+## Sequential Code Blocks
+
+When we have a sequence of code blocks like the following:
+
+````markdown
+```javascript
+const add = (a, b) => a + b;
+```
+```typescript
+const add = (a: number, b: number) => a + b;
+```
+```ruby
+def add(a, b)
+  a + b
+end
+```
+````
+
+Only the code block of the active language is shown.
+So if the active language was TypeScript, the above Markdown outputs:
+
+> ```typescript
+> const add = (a: number, b: number) => a + b;
+> ```
+
+This is useful when showing examples.
+
+Note that the code block must have info string set to a valid language id.
+To find the language id, visit its documentation from the [languages page](/languages/).
+
+## Conditional Rendering
+
+Similarly, conditional rendering based on the active language is supported with code blocks with info strings `if:` and `if-not:`.
+
+These are useful when adding language specific notes.
+
+### `if:languages`
+
+Content of the block is rendered if the active language is in the comma separated list of languages.
+
+````markdown
+```if:javascript,typescript
+Shown _if_ the active language is JavaScript **or** TypeScript.
+```
+```if:ruby
+Shown _if_ the active language is Ruby.
+```
+````
+
+If the active language is Ruby, the above outputs:
+
+> Shown _if_ the active language is Ruby.
+
+### `if-not:languages`
+
+Content of the block is hidden if the active language is in the comma separated list of languages.
+
+````markdown
+```if-not:javascript,typescript
+Hidden _if_ the active language is JavaScript **or** TypeScript.
+```
+```if-not:ruby
+Hidden _if_ the active language is Ruby.
+```
+````
+
+If the active language is Ruby, the above outputs:
+
+> Hidden _if_ the active language is JavaScript **or** TypeScript.
+
```

Comment:
```suggestion
Just like code blocks, language specific blocks needs an empty line before and after them or their group to be rendered correctly.
```

Reply:
I'll include this at the top of the section instead.